static/app/components/globalDrawer/components.tsx

```diff
@@ -0,0 +1,110 @@
+import {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import type {DrawerOptions} from 'sentry/components/globalDrawer/types';
+import SlideOverPanel from 'sentry/components/slideOverPanel';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+export interface DrawerPanelProps extends Required<Omit<DrawerOptions, 'onOpen'>> {
+  children: React.ReactNode;
+  isOpen?: boolean;
+  onOpen?: () => void;
+}
+
+export function DrawerPanel({
+  children,
+  isOpen = false,
+  onClose,
+  onOpen,
+  closeOnOutsideClick = true,
+  closeOnEscapeKeypress = true,
+}: DrawerPanelProps) {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(isOpen);
+  const panelRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(panelRef, () => {
+    if (isOpen && closeOnOutsideClick) {
+      onClose();
+      setIsDrawerOpen(false);
+    }
+  });
+
+  const handleEscapePress = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && isOpen && closeOnEscapeKeypress) {
+        onClose();
+        setIsDrawerOpen(false);
+      }
+    },
+    [isOpen, closeOnEscapeKeypress, onClose]
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleEscapePress);
+    return () => document.removeEventListener('keydown', handleEscapePress);
+  }, [handleEscapePress]);
+
+  useEffect(() => {
+    setIsDrawerOpen(isOpen);
+  }, [isOpen]);
+
+  return (
+    <div role="complementary" aria-hidden={!isDrawerOpen} data-test-id="drawer-content">
+      <SlideOverPanel
+        slidePosition="right"
+        collapsed={!isDrawerOpen}
+        ref={panelRef}
+        onOpen={onOpen}
+      >
+        <DrawerHeader>
+          <CloseButton
+            priority="link"
+            size="zero"
+            borderless
+            data-test-id="close-drawer-control"
```

Comment:
Let's drop this teset id, you can already select this based on it's accessibility attributes.